src/app/services/actions/permissions-action-creator.ts

```diff
@@ -142,3 +148,157 @@ export function consentToScopes(scopes: string[]): Function {
+      dispatch(
+        setQueryResponseStatus({
+          statusText: translateMessage('Default scope'),
+          status: translateMessage('Cannot delete default scope'),
+          ok: false,
+          messageType: MessageBarType.error
+        })
+      );
+      return;
+    }
+
+    if (!userHasRequiredPermissions(requiredPermissions, consentedScopes)) {
+      dispatch(
+        setQueryResponseStatus({
+          statusText: translateMessage('Unable to dissent'),
+          status: translateMessage('You require the following permissions to revoke'),
+          ok: false,
+          messageType: MessageBarType.error
+        })
+      );
+      return;
+    }
+
+    try {
+      if (!consentedScopes && consentedScopes.length < 1) {
+        return;
+      }
+
+      const newScopesArray: string[] = (consentedScopes.filter((scope: string) => scope !== permissionToDelete));
+      const newScopesString = newScopesArray.join(' ');
+
+      const servicePrincipalAppId = await getCurrentAppId(consentedScopes);
+      response = await getPermissionGrant(consentedScopes, servicePrincipalAppId, profile.id);
+      const permissionGrantId = response.id;
+
+      await revokePermission(consentedScopes, permissionGrantId, newScopesString);
+
+      response = await getPermissionGrant(consentedScopes, servicePrincipalAppId, profile.id);
+      const updatedScopes = response.scope.split(' ');
+
+      if (updatedScopes.length !== newScopesArray.length) {
+        return;
+      }
+
+      const authResponse = await getNewAuthObject(updatedScopes);
+
+      if (authResponse && authResponse.accessToken) {
+        dispatch(getAuthTokenSuccess(true));
+        dispatch(getConsentedScopesSuccess(authResponse.scopes));
+        dispatch(
+          setQueryResponseStatus({
+            statusText: translateMessage('Success'),
+            status: translateMessage('Permission revoked'),
+            ok: true,
+            messageType: MessageBarType.success
+          })
+        );
+        if (authResponse.account && authResponse.account.localAccountId !== profile?.id) {
+          dispatch(getProfileInfo());
+        }
+      }
+
+    }
+    catch (error: any) {
+      const { statusCode, code } = error;
+      dispatch(
+        setQueryResponseStatus({
+          statusText: statusCode && code ? (statusCode + ' ' + code) : translateMessage('Unable to dissent'),
+          status: error ? error.message : translateMessage('An error occurred when dissenting'),
+          ok: false,
+          messageType: MessageBarType.error
+        })
+      );
+    }
+  }
+}
+
+const getQuery: IQuery = {
+  selectedVerb: 'GET',
+  sampleHeaders: [
+    {
+      name: 'Cache-Control',
+      value: 'no-cache'
+    }
+  ],
+  selectedVersion: '',
+  sampleUrl: ''
+};
+
+const userHasRequiredPermissions = (requiredPermissions: string[],
+  consentedScopes: string[]) => {
+  return requiredPermissions.every(scope => consentedScopes.includes(scope));
```

Comment:
The functionality here can also be moved to the getPermissionResponse function so that we do not need to import the Graph Client. `getPermissionResponse` can be renamed to something more generic